hongyeollee/app.js

```diff
@@ -33,21 +33,136 @@ app.get("/ping", (req, res) => {
   res.json({ message: "pong" });
 });
 
-//create user
+//과제 2.create user
 app.post("/signup", async (req, res) => {
-  const { name, password, age, email } = req.body;
+  const { name, password, profileImage, email } = req.body;
 
   await mysqlDatabase.query(
     `INSERT INTO users(
           name,
           password,
-          age,
+          profile_image,
           email
         )VALUES(?, ?, ?, ?);
         `,
-    [name, password, age, email]
+    [name, password, profileImage, email]
   );
-  res.status(201).json({ message: "userCreated ✅" });
+  res.status(201).json({ message: "userCreated 👫" });
+});
+
+//과제 3. create post
+app.post("/post", async (req, res) => {
+  const { title, content, imageUrl } = req.body;
+
+  await mysqlDatabase.query(
+    `INSERT INTO posts(
+          title,
+          content,
+          image_url
+        )VALUES(?, ?, ?);
+        `,
+    [title, content, imageUrl]
+  );
+  res.status(201).json({ message: "postCreated 📝" });
+});
+
+//과제 4. search and get posts
+app.get("/getpost", async (req, res) => {
+  await mysqlDatabase.query(
+    `SELECT
+        u.id AS userId,
+        u.profile_image AS userProfileImage,
+        p.id AS postingId,
+        p.image_url AS postingImageUrl,
+        p.content AS postingContent
+    FROM users u
+    INNER JOIN posts p
+    ON u.id=p.id`,
+    (err, rows) => {
+      res.status(200).json({ data: rows });
+    }
+  );
+});
+
+//과제 5. get post of targeting user
+app.get("/postofuser", async (req, res) => {
+  await mysqlDatabase.query(
+    `SELECT
+      u.id AS userId,
+      u.profile_image AS userProfileImage,
+        JSON_ARRAYAGG(
+          JSON_OBJECT(
+            'postingId',p.id,
+            'postingImageUrl', p.image_url,
+            'postingContent', p.content)) AS posting
+    FROM posts p
+    INNER  JOIN users u
+    ON u.id=p.id
+    GROUP BY u.id;
+    `,
+    (err, rows) => {
+      res.status(200).json({ data: rows });
+    }
+  );
+});
+
+// 과제 6. update posting content
+app.patch("/updatecontent", async (req, res) => {
+  const { postContent, postId } = req.body;
+
+  await mysqlDatabase.query(
+    `UPDATE posts p
+     SET
+          p.content=?
+          WHERE p.id=?
+    `,
+    [postContent, postId]
+  );
+
+  await mysqlDatabase.query(
+    `SELECT
+        u.id AS userId,
+        u.name AS userName,
+        p.id AS postingId,
+        p.title AS postingTitle,
+        p.content AS postingContent
+    FROM users u
+    INNER JOIN posts p
+    ON u.id=p.id WHERE p.id=${postId};
+    `,
+    (err, rows) => {
+      res.status(201).json({ data: rows });
+    }
+  );
+});
+
+// 과제 7. delete targeting post
+app.delete("/delete", async (req, res) => {
```

Comment:
홍열님! delete는 메소드 자체에 삭제의 의미를 담고 있습니다. 따라서 게시글을 삭제한다면,
URI`/post`의 delete method라면 게시글을 삭제한다 라는 의미가 조금 더 명확해 질거라고 생각합니다~
```suggestion
app.delete("/post", async (req, res) => {
```

Reply:
post에 대한 행위를 어떻게 진행하느냐에따라서 method만 달라지면 된다는 걸 의미해주시는게 맞을까요?
만약 그렇다면 이해를 제가 하고 있다는 것으로 판단되는데 그게 아니라면 추가적인 설명 부탁드리겠습니다.